src/dir.rs

```diff
@@ -305,8 +305,12 @@ impl<'a, IO: ReadWriteSeek, TP: TimeProvider, OCC: OemCpConverter> Dir<'a, IO, T
                 let sfn_entry = self.create_sfn_entry(dot_sfn, FileAttributes::DIRECTORY, entry.first_cluster());
                 dir.write_entry(".", sfn_entry)?;
                 let dotdot_sfn = ShortNameGenerator::generate_dotdot();
-                let sfn_entry =
-                    self.create_sfn_entry(dotdot_sfn, FileAttributes::DIRECTORY, self.stream.first_cluster());
+                let dotdot_cluster = if self.stream.first_cluster() == self.fs.root_dir().first_cluster() {
```

Comment:
I think checking if we are in the root directory should be a simple operation. Calling `fs.root_dir()`  feels like an overkill to me.
I propose to add `DirRawStream::is_root_dir` which would call `File::is_root_dir` which would check if `File::entry == None`

Reply:
Yeah, that sounds like a better solution. Admittedly mine was more of a "first fit" thing.